### PR TITLE
[ACL-138] Add support for retry parameter

### PR DESCRIPTION
**Pull request by tl-luca-baggi**
# Description

Add support for `retry` parameter:
- customers can opt in to retry feature initialising the `retry` property in the `CreatePayment.PaymentMethod.BankTransfer` object
- added support for new `AttemptFailed` payment details 
- added unit, integration and acceptance tests

## Type of change

Please select multiple options if required.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code where necessary
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the relevant documentation

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Main properties
 group=com.truelayer
 archivesBaseName=truelayer-java
-version=13.0.5
+version=13.1.0
```
Comment (tl-luca-baggi) [inline gradle.properties:4]: Setting this as a minor change as it shouldn't break any existing integration, but given the feature itself I'm ok if we want to release a new major version

File: src/main/java/com/truelayer/java/payments/entities/paymentdetail/AttemptFailedPaymentDetail.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    AuthorizationFlowWithConfiguration authorizationFlow;
+
+    @JsonGetter
+    public Optional<AuthorizationFlowWithConfiguration> getAuthorizationFlow() {
+        return Optional.ofNullable(authorizationFlow);
+    }
+
+    @RequiredArgsConstructor
+    @Getter
+    public enum FailureStage {
```
Comment (tl-luca-baggi) [inline src/main/java/com/truelayer/java/payments/entities/paymentdetail/AttemptFailedPaymentDetail.java:37]: With respect to `Failed`, this new status can be reached only from `authorizing` and `authorized` (see [here](https://docs.truelayer.com/docs/web-sdk#payment-retries-for-web-sdk))

File: src/test/java/com/truelayer/java/TestUtils.java [hunk 3]
```diff
@@ -285,13 +283,46 @@ public static void runAndAssertHeadlessResourceAuthorisation(
...
+        REJECT_AUTHORISATION("RejectAuthorisation");
 
-        private final String path;
-        private final String payload;
+        private final String action;
+    }
+
+    @RequiredArgsConstructor
+    @Getter
+    public static class HeadlessResourceAuthorization {
```
Comment (tl-luca-baggi) [inline src/test/java/com/truelayer/java/TestUtils.java:303]: Refactored from enum to class with a builder to allow more flexibility, i.e. allow to select the `action` as well

File: gradle.properties (hunk 1 above)
Comment (dili91) [reply gradle.properties:4]: Minor looks good to me!

File: src/main/java/com/truelayer/java/payments/entities/paymentdetail/Retry.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+import java.util.List;
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class Retry extends AuthorizationFlowAction {
+    Type type = Type.RETRY;
+
+    List<String> retryOptions;
```
Comment (dili91) [inline src/main/java/com/truelayer/java/payments/entities/paymentdetail/Retry.java:12]: Do we want to model this as a list of enum? We usually follow this approach (and I've pinned for rework in our gap analysis where we don't). Example [here](https://github.com/TrueLayer/truelayer-java/blob/b5ba51ba559ff0d8b14776461b5dc320a18e73e7/src/main/java/com/truelayer/java/entities/ProviderFilter.java#L21)

File: src/test/java/com/truelayer/java/TestUtils.java [hunk 5]
```diff
@@ -285,13 +283,46 @@ public static void runAndAssertHeadlessResourceAuthorisation(
...
+        final HeadlessResourceAction action;
+        final HeadlessResource resource;
+        String path;
+        String payload;
```
Comment (dili91) [inline src/test/java/com/truelayer/java/TestUtils.java:307]: nit: make private

File: src/test/java/com/truelayer/java/acceptance/MandatesAcceptanceTests.java [hunk 6]
```diff
@@ -276,7 +288,13 @@ public void itShouldCreateAndRevokeAMandate(String mandatesScope, Mandate.Type m
                 .getNext()
                 .asRedirect()
                 .getUri();
-        runAndAssertHeadlessResourceAuthorisation(tlClient, redirectUri, HeadlessResourceAuthorization.MANDATES);
+        runAndAssertHeadlessResourceAuthorisation(
```
Comment (dili91) [inline src/test/java/com/truelayer/java/acceptance/MandatesAcceptanceTests.java:291]: nice!

**Review by dili91 — COMMENTED** [review]
LGTM 🥳  - Just left a couple of minor comments. Happy to evaluate the addition of the new HPP link builder here already, via opt-in. If we think it's useful

File: src/test/java/com/truelayer/java/TestUtils.java (hunk 5 above)
Comment (tl-luca-baggi) [reply src/test/java/com/truelayer/java/TestUtils.java:307]: Good catch

File: src/main/java/com/truelayer/java/payments/entities/paymentdetail/RetryOption.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
+package com.truelayer.java.payments.entities.paymentdetail;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+@Getter
+public enum RetryOption {
+    RESTART("restart");
```
Comment (tl-luca-baggi) [inline src/main/java/com/truelayer/java/payments/entities/paymentdetail/RetryOption.java:10]: This is currently the only option available (confirmed by @tl-Roberto-Mancinelli )

**Review by dili91 — APPROVED** [review]

**Review by dili91 — APPROVED** [review]

**Pull request merged**